Code/GraphMol/Fingerprints/MHFP.h

```diff
@@ -0,0 +1,302 @@
+  //! Constructor
+  /*!
+    \brief Construct a MHFPEncoder
+   
+    The MHFPEncoder class is instantieted with a given number of permutations
+    and a seed. Fingerprints / minhashes created with a different number of
+    permutations or a different seed are not compatible.
+   
+    \param n_permutations the number of permutations used to create hash functions.
+              This will be the dimensionality of the resulting vector.
+              Default: <tt>2048</tt>.
+    \param seed a random seed. Default: <tt>42</tt>.
+   */
+  MHFPEncoder(unsigned int n_permutations = 2048,
+        unsigned int seed = 42);
+  
+  /*!
+    \brief Creates a MinHash from a vector of strings.
+   
+    This method is exposed in order to enable advanced usage of this MHFP 
+    implementation such as customizing the properties that are hashed in order 
+    to create an MHFP instance. In theory, any number of values that can be 
+    represented  as strings can be minhashed. This method is called 
+    by MHFPEncoder::Encode.
+   
+    \param vec a vector containg strings (e.g. the smiles shingling of a molecule).
+
+    \returns the MinHash of the input.
+   */
+  std::vector<uint32_t>
+  FromStringArray(const std::vector<std::string>& vec);
+
+  /*!
+    \brief Creates a MinHash from a list of unsigned integers.
+   
+    This method is exposed in order to enable advanced usage of this MHFP 
+    implementation such as MinHashing a sparse array generated by another
+    fingerprint (e.g. Morgan / ECFP).
+   
+    \param vec a vector containg unsigned integers.
+
+    \returns the MinHash of the input.
+   */
+  std::vector<uint32_t>
+  FromArray(const std::vector<uint32_t>& vec);
+
+  /*!
+    \brief Creates a molecular shingling based on circular substructures.
+   
+    A molecular shingling is a vector of SMILES that were extracted from and
+    represent a molecule. This method extracts substructures centered at each
+    atom of the molecule with different radii. A molecule with 10 atoms will
+    generate <tt>10 * 3</tt> shingles when a radius of <tt>3</tt> is chosen.
+   
+    \param radius the maximum radius of the substructure that is generated at
+          each atom. Default: <tt>3</tt>.
+    \param rings whether the rings (SSSR) are extrected from the molecule and
+           added to the shingling. Given the molecule 
+           <tt>"C1CCCCCC1C(=O)C"</tt>, "<tt>C1CCCCCC1"</tt> would be added
+           to the shingling. Default: <tt>true</tt>.
+    \param isomeric whether the SMILES added to the shingling are isomeric.
+            Default: <tt>false</tt>.
+    \param kekulize whether the SMILES added to the shingling are kekulized.
+            Default: <tt>true</tt>.
+    \param min_radius the minimum radius that is used to extract n-grams.
+            Default: <tt>1</tt>.
+
+    \returns the shingling of a molecule.
+   */
+  std::vector<std::string>
+  CreateShingling(const ROMol& mol,
+          const unsigned char& radius = 3,
```

Comment:
Taking const references to POD types like char, bool, int, double, etc. is redundant. Please replace all of these with just the pure data type. So this argument would become: `unsigned char radius = 3`.
For more info about this and an explanation: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in